visidata/main.py

```diff
@@ -129,26 +131,43 @@ def outputProgressEvery(vd, sheet, seconds:float=0.5):
         time.sleep(seconds)
 
 @visidata.VisiData.api
-def moveToPos(vd, sources, startsheets, startrow, startcol):
-    sheets = []  # sheets to apply startrow:startcol to
+def moveToPos(vd, sources, startsheets, startcol, startrow):
+    sheets = []  # sheets to apply startcol:startrow to
     if not startsheets:
-        sheets = sources  # apply row/col to all sheets
+        sheets = sources  # apply col/row to all sheets
+        for vs in sheets:
+            vd.sync(vs.ensureLoaded())
```

Comment:
So, the problem with this sync, is that opening a large file with a start position specified, will wait until the entire file has loaded, before it shows any interface at all.  (Prior to this change, it loads the file immediately and reports that it can't find the column).

Neither behavior is what the user wants, but the current behavior is at least immediately responsive, whereas adding a sync() here will appear to delay startup indefinitely, with no feedback.

My proposed behavior, would be to wait a second or two after loading on the sheet has started (or until loading has finished, if that's sooner), before trying to move the cursor.  This will fix the issue in most cases, since columns are usually discovered very early in the loading process.  If the column hasn't been discovered yet, it will fail as it does currently.  Alternatively, we could start a thread that keeps trying to move the cursor to the given starting position until it either succeeds, or loading has finished.

Reply:
Good point, responsiveness is critical. I'll think about how to implement it. I do like the idea of a thread that keeps trying. Because then the delay before the cursor move could be shorter than a second, and unobtrusive.